Sources/Danger/Plugins/SwiftLint/SwiftLint.swift

```diff
@@ -59,19 +59,24 @@ extension SwiftLint {
             violations = makeViolations(from: outputJSON, failAction: failAction)
         } else {
             // Gathers modified+created files, invokes SwiftLint on each, and posts collected errors+warnings to Danger.
-            var files = danger.git.createdFiles + danger.git.modifiedFiles
+            var files = (danger.git.createdFiles + danger.git.modifiedFiles).filter { $0.hasSuffix(".swift") }
             if let directory = directory {
                 files = files.filter { $0.hasPrefix(directory) }
             }
 
-            violations = files.filter { $0.hasSuffix(".swift") }.flatMap { file -> [SwiftLintViolation] in
-                var arguments = ["lint", "--quiet", "--path \"\(file)\"", "--reporter json"]
-                if let configFile = configFile {
-                    arguments.append("--config \"\(configFile)\"")
-                }
-                let outputJSON = shellExecutor.execute(swiftlintPath, arguments: arguments)
-                return makeViolations(from: outputJSON, failAction: failAction)
+            // swiftlint takes input files in the format of `SCRIPT_INPUT_FILE_COUNT=2 SCRIPT_INPUT_FILE_0="file1.swift" SCRIPT_INPUT_FILE_1="file2.swift" swiftlint lint`
+            var inputFiles = "SCRIPT_INPUT_FILE_COUNT=\(files.count)"
+            for (index, file) in files.enumerated() {
+                inputFiles.append(" SCRIPT_INPUT_FILE_\(index)=\"\(file)\"")
             }
+
+            var arguments = ["lint", "--quiet", "--use-script-input-files", "--force-exclude", "--reporter json"]
+            if let configFile = configFile {
+                arguments.append("--config \"\(configFile)\"")
+            }
+
+            let outputJSON = shellExecutor.execute([inputFiles, swiftlintPath].joined(separator: " "), arguments: arguments)
```

Comment:
<!--
  0 failure: 
  1 warning:  Line should be 12...
  
  
  DangerID: danger-id-default;
  File: Sources/Danger/Plugins/SwiftLint/SwiftLint.swift;
  Line: 78;
-->

- :warning: Line should be 120 characters or less: currently 124 characters (`line_length`)